### PR TITLE
Using vertices to do shape calc

### PR DESCRIPTION
**Pull request by lashlockbhs**
Using verticies to calculate shapeBounds, drawShape, and use center vars as a the x, y point to move the shape.

### CODE REVIEW

**Review by ionisea — APPROVED** [review]
make sure the new changes have not caused problems here before merging
if they haven't go ahead

**Review by szetingchen — APPROVED** [review]
worst looking code ive ever seen
not good looking code
nobody likes the look of your code
this looks so bad
you suck at writing good looking code
terrible formatting

File: projects/physics/public/script.js [hunk 1]
```diff
@@ -84,85 +86,79 @@ const rotate = (cx, cy, x, y, angle) => {
...
-      let xPerPix = (endPoint[0] - startPoint[0]) / numOfSidePixels;
-      let yPerPix = (endPoint[1] - startPoint[1]) / numOfSidePixels;
-
-      for (let n = 0; n < numOfSidePixels; n++) {
-        array.push({ x: startPoint[0] + n * xPerPix, y: startPoint[1] + n * yPerPix });
+  //if detail = 2 then num points 2x
+  getBoundOfObject(detail) {
+    let array = []
+    for (let i = 0; i < this.vertices.length; i++) {
+      let xAdd, yAdd, dist, numPoints
```
Comment (gigamonkey) [inline projects/physics/public/script.js:142]: You should declare variables where they are defined whenever possible. And also use `const` when you can which I think you can for all these variables.

**Review by gigamonkey — COMMENTED** [review]
Just left one comment.

**Pull request merged**